### PR TITLE
ENH: Add `HASI` Python module

### PR DESCRIPTION
**Pull request by tbirdso**
- Moved Python scripts into `src/hasi/hasi` module
- Added `pypackage.toml` with `flit`
- Updated pytests with new locations

Will close [Issue 33](https://github.com/KitwareMedical/HASI/issues/33).

### CODE REVIEW

**Comment by tbirdso** [comment]
Check failures are already tracked in [Issue 30](https://github.com/KitwareMedical/HASI/issues/30). This PR is ready for review.

File: src/hasi/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+[build-system]
+requires = ["flit_core >=2,<4"]
+build-backend = "flit_core.buildapi"
+
+[tool.flit.metadata]
+module = "hasi"
+author = "Kitware Medical"
+author-email = "matt.mccormick@kitware.com"
+home-page = "https://github.com/KitwareMedical/HASI"
+classifiers = ["License :: OSI Approved :: Apache Software License"]
```
Comment (thewtex) [inline src/hasi/pyproject.toml:10]: Please add a few more classifiers from:

https://github.com/InsightSoftwareConsortium/ITKPythonPackage/blob/0b64128a0ee699d35a3e4c971f748743c4d50106/setup.py#L51-L67

**Review by thewtex — CHANGES_REQUESTED** [review]
@tbirdso very nice, thanks!

Next steps:

- Add *itk-hasi* to the*hasi* package dependencies in pyproject.toml
- Move the *test/test_meshtomeshregistration.py* into *src/hasi/*, add pytest to the pypackage.toml testing dependencies, and ensure we can run *pytest* to run the test from *src/hasi*.
- Add a new workflow in *.github/workflows/* to test the *hasi* package. It should cd into the directory, install dependencies, run the tests, and build the package.

**Comment by tbirdso** [comment]
@thewtex Ready for review. The `test-python-hasi-package` runs `pytest` successfully on `hasi` Python registration scripts.

**Review by thewtex — APPROVED** [review]
@tbirdso excellent! :1st_place_medal:

**Pull request merged**